### PR TITLE
Add custom annotations support for catalog sync and connectInject pods via Helm

### LINKED ISSUE #760: Custom annotations for catalog sync, connectInject, and mesh gateway pods
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request. Searching for pre-existing feature requests helps us consolidate datapoints for identical requirements into a single place, thank you!
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

---

#### Feature Description

Annotations like labels provide additional functionality that are exposed via third party mutating web hooks or attach valuable metadata that is not applicable via labels (i.e. values with certain symbols like `@`)

- connect inject
- catalog sync
- mesh gw

ACL Init is already filed as a separate issue here: - https://github.com/hashicorp/consul-helm/issues/543

### PR DESCRIPTION
**Pull request by jeanmorais**
Changes proposed in this PR:
- Add custom annotations support for catalog sync and connectInject pods via Helm.

How I've tested this PR:
- Unit tests

Fixes #760

How I expect reviewers to test this PR:
* `bats sync-catalog-deployment.bats`
* `bats connect-inject-deployment.bats`

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/connect-inject-deployment.yaml [hunk 1]
```diff
@@ -51,7 +54,6 @@ spec:
         {{- if .Values.global.secretsBackend.vault.agentAnnotations }}
         {{ tpl .Values.global.secretsBackend.vault.agentAnnotations . | nindent 8 | trim }}
         {{- end }}
-        {{- end }}
```
Comment (lkysow) [inline charts/consul/templates/connect-inject-deployment.yaml:54]: why is this being removed?

File: charts/consul/templates/sync-catalog-deployment.yaml [hunk 2]
```diff
@@ -46,7 +49,6 @@ spec:
         {{- if .Values.global.secretsBackend.vault.agentAnnotations }}
         {{ tpl .Values.global.secretsBackend.vault.agentAnnotations . | nindent 8 | trim }}
         {{- end }}
-        {{- end }}
```
Comment (lkysow) [inline charts/consul/templates/sync-catalog-deployment.yaml:49]: same here

**Review by lkysow — CHANGES_REQUESTED** [review]
LGTM but I think there are some `{-end}`'s being deleted incorrectly

File: charts/consul/templates/connect-inject-deployment.yaml (hunk 1 above)
Comment (jeanmorais) [reply charts/consul/templates/connect-inject-deployment.yaml]: It went unnoticed. Fixed.

File: charts/consul/templates/sync-catalog-deployment.yaml (hunk 2 above)
Comment (jeanmorais) [reply charts/consul/templates/sync-catalog-deployment.yaml]: fixed

**Review by lkysow — APPROVED** [review]
Thank you so much!

**Pull request merged**